spec/api_spec.rb

```diff
@@ -172,6 +172,50 @@ def write_token token
     end
   end
 
+  describe "#username" do
+    let(:jwt_payload) do
+      'eyJzdWIiOiJ1c2VyLTlhYjBiYmZiOWJlNjA5Yzk2ZjUyN2Y1YiIsImlhdCI6MTYwMzQ5MDA4MH0='
+    end
+    let(:jwt_header) do
+      'eyJhbGciOiJjb25qdXIub3JnL3Nsb3NpbG8vdjIiLCJraWQiOiI2MWZjOGRiZDM4MjA4NDll' \
+      'ZDI4YTZhYTAwMzFjNjM5MjkxZjJmMDQzNDVjYTU0MWI5NzUxMGQ5NjkyM2I3NDlmIn0='
+    end
+    let(:conjur_token) do
+      {
+        'data' => 'conjur-user-1234',
+        'timestamp' => Time.now.to_s
+      }
+    end
+    let(:jwt_token) do
+      {
+        'protected' => jwt_header,
+        'payload' => jwt_payload,
+      }
+    end
+
+    it "can correctly extract the username from old Conjur token" do
+      expect(Conjur::API.new_from_token(conjur_token).username).to \
+        eq('conjur-user-1234')
+    end
+
+    context 'when using JWT token' do
+      it "can correctly extract username" do
+        expect(Conjur::API.new_from_token(jwt_token).username).to \
+          eq('user-9ab0bbfb9be609c96f527f5b')
+      end
```

Comment:
I love that your maintaining nice clean margins.  Another formatting suggestion, take it or leave it: Once your `to` has to break lines, give it parens. So here it would become:
```ruby
        expect(Conjur::API.new_from_token(jwt_token).username).to(
          eq('user-9ab0bbfb9be609c96f527f5b')
        )